daemons/attrd/attrd_sync.c

```diff
@@ -149,37 +146,41 @@ attrd_remove_client_from_waitlist(pcmk__client_t *client)
 }
 
 void
-attrd_ack_waitlist_clients(enum attrd_sync_point sync_point, const char *attr_name)
+attrd_ack_waitlist_clients(enum attrd_sync_point sync_point, const xmlNode *xml)
 {
-    GHashTableIter iter;
+    int callid;
     gpointer value;
 
     if (waitlist == NULL) {
         return;
     }
 
-    g_hash_table_iter_init(&iter, waitlist);
+    if (crm_element_value_int(xml, XML_LRM_ATTR_CALLID, &callid) == -1) {
+        crm_warn("Could not get callid from request XML");
+        return;
```

Comment:
This will get called for all requests, not just those needing acks, right? So probably just a trace if not present

Reply:
Figuring out a way to skip all this for attributes that don't need an ACK is on my todo list.  I imagine multiple function calls for every single attribute update is not going to be great for performance.